libgalois/include/katana/analytics/k_truss/k_truss.h

```diff
@@ -0,0 +1,72 @@
+#ifndef KATANA_LIBGALOIS_KATANA_ANALYTICS_KTRUSS_KTRUSS_H_
+#define KATANA_LIBGALOIS_KATANA_ANALYTICS_KTRUSS_KTRUSS_H_
+
+#include <iostream>
+
+#include <katana/analytics/Plan.h>
+#include "katana/analytics/Utils.h"
+
+// API
+
+namespace katana::analytics {
+
+/// A computational plan to for k-truss, specifying the algorithm and any
+/// parameters associated with it.
+class KTrussPlan : Plan {
+public:
+  /// Algorithm selectors for KCore
+  enum Algorithm { kBsp, kBspJacobi, kBspCoreThenTruss };
+
+  // Don't allow people to directly construct these, so as to have only one
+  // consistent way to configure.
+private:
+  Algorithm algorithm_;
+
+  KTrussPlan(Architecture architecture, Algorithm algorithm)
+      : Plan(architecture), algorithm_(algorithm) {}
+
+public:
+  KTrussPlan() : KTrussPlan{kCPU, kBsp} {}
+
+  Algorithm algorithm() const { return algorithm_; }
+
+  /// Bulk-synchronous parallel algorithm.
+  static KTrussPlan Bsp() { return {kCPU, kBsp}; }
+
+  /// Bulk-synchronous parallel with separated edge removal algorithm.
+  static KTrussPlan BspJacobi() { return {kCPU, kBspJacobi}; }
+
+  /// Compute k-1 core and then k-truss algorithm.
+  static KTrussPlan BspCoreThenTruss() { return {kCPU, kBspCoreThenTruss}; }
+};
+
+/// Compute the k-truss for pfg. The pfg is expected to be
+/// symmetric.
```

Comment:
Nit: Not "expected to be symmetric" doesn't really express the reality that the algorithm will produce wrong results if the graph isn't symmetric. What about "The graph must be symmetric."